lib/jsdom/browser/Window.js

```diff
@@ -403,6 +404,21 @@ function matchesDontThrow(el, selector) {
 }
 
 function startTimer(window, startFn, stopFn, callback, ms) {
+  if (typeof callback !== "function") {
+    const code = String(callback);
+    callback = window._globalProxy.eval.bind(window,
+      `(function(){` + code + `}())\n//# sourceURL=${window.location.href}`);
```

Comment:
I wanted to make sure `arguments.callee.toString` doesn't include the sourceURL, but executed code can't access arguments anyways, so removed.